algorithms.tex

```diff
@@ -8,6 +8,220 @@
 
 \subsection{Numerics}
 
+The \texttt{Float} class holds an arbitrary-precision binary floating-point value
+and a precision in bits. An operation between two \texttt{Float}
+inputs is rounded to the larger of the two precisions.
+
+The preferred way to evaluate an expression numerically is with the
+\texttt{evalf} method, which internally estimates the number of accurate
+bits of the floating-point
+approximation for each sub-expression, and adaptively increases the
+working precision until the estimated accuracy of the
+final result matches the sought number of decimal digits.
+
+The internal error tracking does not provide rigorous error bounds
+(in the sense of interval arithmetic) and cannot be used to track
+uncertainty in measurement data in any meaningful way;
+the sole purpose is to mitigate loss of accuracy that typically occurs
+when converting symbolic expressions to numerical values, for example
+due to catastrophic cancellation. This is illustrated by the following
+example (the input 25 specifies that 25 digits are sought):
+
+\begin{verbatim}
+>>> cos(exp(-100)).evalf(25) - 1
+0
+>>> (cos(exp(-100)) - 1).evalf(25)
+-6.919482633683687653243407e-88
+\end{verbatim}
+
+The \texttt{evalf} method works with complex numbers and supports
+more complicated expressions, such as
+special functions, infinite series and integrals.
+
+SymPy does not track the accuracy of
+approximate numbers outside of \texttt{evalf}.
+The familiar dangers of floating-point arithmetic apply, and 
+symbolic expressions containing floating-point numbers should be treated
+with some caution.
+This approach is similar to Maple and Maxima.
```

Comment:
We will at least need general citations for Maple and Maxima. If there are more general ones for the fact that they use this approach to floating point numbers that would be nice as well.